src/cocotb/_scheduler.py

```diff
@@ -421,6 +423,8 @@ def _queue(
         # Don't queue the same task more than once (gh-2503)
         if task in self._pending_tasks:
             raise InternalError("Task was queued more than once.")
+        # TODO Move state tracking into Task
+        task._state = Task._State.SCHEDULED
         self._pending_tasks[task] = outcome
```

Comment:
Rename `_pending_tasks` to reflect `SCHEDULED` state, if tasks waiting on a `Trigger` are in the `PENDING` state.